SixTrack/sixtrack.s

```diff
@@ -7393,269 +7437,302 @@ cc2008
+    
+! 1 shift
+      IF (wire_flagco(ix).eq.1) THEN
+         xi = (xv(1,j)+dx)*c1m3 !SI
+         yi = (xv(2,j)+dy)*c1m3 !SI
+      ELSE IF (wire_flagco(ix).eq.-1) THEN
+         xi = (xv(1,j)+( dx-wire_clo(1,wire_num(i)) ))*c1m3 !SI
+         yi = (xv(2,j)+( dy-wire_clo(2,wire_num(i)) ))*c1m3 !SI
+      END IF 
+
+! x'-> px; y'->py
+      yv(1,j) = yv(1,j)*(1d0 + dpsv(j))
+      yv(2,j) = yv(2,j)*(1d0 + dpsv(j))
+
+! start not crlibm wire
 +if .not.crlibm
-      yv(2,j) = sqrt((1d0+dpsv(j))**2-yv(1,j)**2)*sin(atan(yv(2,j)/     &!hr03
-     &sqrt(((1d0+dpsv(j))**2-yv(1,j)**2)-yv(2,j)**2))+ty)                !hr03
-+ei
 
-+if crlibm
-      xv(2,j) = xv(2,j)-(((xv(1,j)*sin_rn(-1d0*tx))*yv(2,j))/           &!hr03
-     &sqrt((1d0+dpsv(j))**2-yv(2,j)**2))/cos_rn(atan_rn(yv(1,j)/        &!hr03
-     &sqrt(((1d0+dpsv(j))**2-yv(1,j)**2)-yv(2,j)**2))+tx)                !hr03
-+ei
-+if .not.crlibm
-      xv(2,j) = xv(2,j)-(((xv(1,j)*sin(-1d0*tx))*yv(2,j))/              &!hr03
-     &sqrt((1d0+dpsv(j))**2-yv(2,j)**2))/cos(atan(yv(1,j)/              &!hr03
-     &sqrt(((1d0+dpsv(j))**2-yv(1,j)**2)-yv(2,j)**2))+tx)                !hr03
-+ei
-!+if crlibm
-!+ei
-!+if .not.crlibm
-!+ei
-+if crlibm
-      xv(1,j) = xv(1,j)*                                                &!hr03
-     &(cos_rn(-1d0*tx)-sin_rn(-1d0*tx)*tan_rn(atan_rn(yv(1,j)/          &!hr03
-     &sqrt(((1d0+dpsv(j))**2-yv(1,j)**2)-yv(2,j)**2))+tx))               !hr03
-+ei
-+if .not.crlibm
-      xv(1,j) = xv(1,j)*(cos(-1d0*tx)-sin(-1d0*tx)*tan(atan(yv(1,j)/    &!hr03
-     &sqrt(((1d0+dpsv(j))**2-yv(1,j)**2)-yv(2,j)**2))+tx))               !hr03
-+ei
-+if crlibm
-      yv(1,j) = sqrt((1d0+dpsv(j))**2-yv(2,j)**2)*                       !hr03
-     &sin_rn(atan_rn(yv(1,j)/                                            !hr03
-     &sqrt(((1d0+dpsv(j))**2-yv(1,j)**2)-yv(2,j)**2))+tx)                !hr03
-+ei
-+if .not.crlibm
-      yv(1,j) = sqrt((1d0+dpsv(j))**2-yv(2,j)**2)*sin(atan(yv(1,j)/     &!hr03
-     &sqrt(((1d0+dpsv(j))**2-yv(1,j)**2)-yv(2,j)**2))+tx)                !hr03
-+ei
+! ibeco = 0
+      if(ibeco.eq.0) then
+! 2 symplectic rotation of coordinate system (tx, ty)
+          yi = yi-(((xi*sin(tx))*yv(2,j))/                              &
+     &sqrt((1d0+dpsv(j))**2-yv(2,j)**2))/                               &
+     &cos(atan(yv(1,j)/sqrt(((1d0+dpsv(j))**2-yv(1,j)**2)-              &
+     &yv(2,j)**2))-tx)                                                  
+          xi = xi*(cos(tx)-sin(tx)*tan(atan(yv(1,j)/                    &
+     &sqrt(((1d0+dpsv(j))**2-yv(1,j)**2)-yv(2,j)**2))-tx))              
+          yv(1,j) = sqrt((1d0+dpsv(j))**2-yv(2,j)**2)*                  &
+     &sin(atan(yv(1,j)/                                                 &
+     &sqrt(((1d0+dpsv(j))**2-yv(1,j)**2)-yv(2,j)**2))-tx)
+
+          xi = xi-(((yi*sin(ty))*yv(1,j))/                              &
+     &sqrt((1d0+dpsv(j))**2-yv(1,j)**2))/                               &
```

Comment:
FYI, you may loose some numerical precission by writing a^2-b^2, (a+b)*(a-b) is better.